src/test/run-pass/yk_swt/collect-traces-threaded.rs

```diff
@@ -25,29 +25,32 @@ fn main() {
         start_tracing();
         let _ = work2();
         let raw_trace2 = stop_tracing().unwrap();
-        let trace2 = trace_to_vec(&raw_trace2);
-        unsafe { libc::free(raw_trace2.buf() as *mut libc::c_void) };
+        let trace2 = trace_to_vec(raw_trace2);
         trace2
     });
 
     start_tracing();
     black_box(work1());
     let raw_trace1 = stop_tracing().unwrap();
-    let trace1 = trace_to_vec(&raw_trace1);
-    unsafe { libc::free(raw_trace1.buf() as *mut libc::c_void) };
+    let trace1 = trace_to_vec(raw_trace1);
 
     let trace2 = thr2.join().unwrap();
 
     assert_ne!(trace1, trace2); // They should have been thread local.
 }
 
 // Copies a trace into a plain Rust Vec of tuples so we can compare them.
-fn trace_to_vec(t: &SWTrace) -> Vec<(u64, u32, u32)> {
+fn trace_to_vec(tup: (*mut SWTMirLoc, usize)) -> Vec<(u64, u32, u32)> {
+    let (buf, len) = tup;
     let mut v = Vec::new();
-    for i in 0..t.len() {
-        let loc = t.loc(i);
+    for i in 0..len {
+        if i > isize::max_value() as usize {
```

Comment:
Wouldn't it be easier to hoist this outside the loop? Something like `assert!(len < (isize::max_value() as usize))`.

Reply:
Indeed!